adr/0018-public-ids-for-qualifications.md

```diff
@@ -0,0 +1,70 @@
+# 17. Application Choice Updated At
+
+Date: 2021-01-28
+
+## Status
+
+In progress
+
+## Context
+
+The Vendor API exposes candidate qualifications and grades for systems to ingest.
+Each qualification in the system has an id, which vendor systems use to identify the qualification.
+It is therefore important that each qualification coming from the API has a unique id.
+We use the database id for a qualification object for this purpose.
+
+An issue arises when we have qualifications that are made up of multiple exams/grades.
+See [the ADR on structured grades](./0016-structured-gcse-grades.md) for background on the way gcse qualifications are stored.
+API consumers expect one qualification entity per exam/grade.
+However, because of the structured grades introduced above, some exams/grades share the same database id.
+Therefore, not all the ids for qualifications coming through the API are unique.
+
+## Decision
+
+We initially considered changing the data type of the id field in the API response to a string, to support ids of the form `123`, `123_1`, `123_2`.
+Vendors were not able to support that change, so the field had to remain an integer.
+
+The solution for this will be split into two: one for qualifications with a simple grade, and one for qualifications with structured grades.
```

Comment:
This isn't true!

Triple science GCSEs are treated like simple grade qualifications in the API even though they have structured grades. This type of GCSE should get a top level public_id, rather than separate public_ids for each grade

Reply:
This was one of the sticking points I wanted to revisit. I'm glad you picked it up, cheers Ben!